linera-core/src/client.rs

```diff
@@ -460,6 +469,34 @@ where
                     }
                 }
             }
+            Reason::NewRound { height, round } => {
+                let chain_id = notification.chain_id;
+                if let Some(info) =
+                    Self::get_local_chain_info(this.clone(), chain_id, &mut local_node).await
+                {
+                    if info.next_block_height > height || info.manager.next_round() > round {
```

Comment:
This could skip `(info.next_block_height + 1, 0)`. Don't we want `(info.next_block_height, info.manager.next_round()) > (height, round)` here?

Actually, is it a good idea to separate `NewRound` from `NewBlock`? I don't think we guarantee that notifications arrive in order.

Reply:
Ah, good catch! :facepalm: 

Yes, we could remove `NewBlock` now—it's implied by a `NewRound` for round 0.

But many clients will only be interested in new blocks, not new rounds, so it could be convenient to keep them separate. It shouldn't hurt that they don't arrive in order?

Reply:
It creates a bit of duplication for those clients who actually care about rounds but ok.